### ISSUE TITLE
Temporary cookies are not supported

### ISSUE THREAD

**Issue by alosultan** [scrapinghub/splash#612]
Splash can send cookies to the target server by request headers (args.headers), but does not save them during the execution of the request. If the loaded page contains AJAX code that finds and sends the available cookies to the target server (to load other part of data that is available only for authorized users), then the AJAX code can not find these cookies (they are no longer in the browser), so part of the data will not be available.

This can be corrected if we use the "execute" endpoint and manually set "add _cookie" for each cookie to each request.

**Comment by kmike** [comment]
Yeah; the reason is that endpoints other than /execute started to get less attention when /execute was added. 

Either adding "cookies" argument with the same semantics as [splash:init_cookies](http://splash.readthedocs.io/en/stable/scripting-ref.html#splash-init-cookies), or adding a special handling of cookies inside 'headers' argument for all endpoints except for /execute (similar to how user-agent is handled) sounds fine to me. 

It is unlikely I'll find time soon to fix this issue myself though, so contributions are welcome.

**Comment by alosultan** [comment]
Thank you Mike for your convincing comment and your activity.
